engine/src/main/java/com/datatorrent/stram/client/StramAppLauncher.java

```diff
@@ -535,8 +534,12 @@ public void runLocal(AppFactory appConfig) throws Exception
 
   public URLClassLoader loadDependencies()
   {
-    URLClassLoader cl = URLClassLoader.newInstance(launchDependencies.toArray(new URL[launchDependencies.size()]));
+    URL[] dependencies = launchDependencies.toArray(new URL[launchDependencies.size()]);
+
+    ClassLoader currentContextClassLoader = Thread.currentThread().getContextClassLoader();
```

Comment:
The issue to be addressed here it that additional dependencies need to be added to the existing class loader, hence chaining with previous class loader. Avoiding repeated instantiation with exactly same dependencies is an optimization, first we should get the functional issue addressed.